spock-core/src/main/java/org/spockframework/runtime/SpockNode.java

```diff
@@ -126,4 +130,34 @@ protected static ExecutionMode toExecutionMode(org.spockframework.runtime.model.
       .map(er -> new ExclusiveResource(er.getKey(), toLockMode(er.getMode())))
       .collect(Collectors.toSet());
   }
+
+  @Override
+  public Set<TestTag> getTags() {
+    // Lazy initialisation
+    if (tags == null) {
```

Comment:
The latter consideration is why I decided to cache it. I was afraid to slow down Spock too much just because of tags, not being very familiar with its code base and not knowing how often certain methods might be called.

Reply:
Yes, I have. For every class or method/iteration, `getTags()` is called 3x, once by `org.junit.platform.launcher.TagFilter.includeMatching` and again by `excludeMatching`, then later by `org.junit.platform.launcher.TestIdentifier.from`.

Reply:
I am taking the lack of feedback here as consent for lazy initialisation and caching, resolving this conversation.